docs/manual/common/concepts/ServiceDiscovery.md

```diff
@@ -0,0 +1,39 @@
+# Service Discovery
+
+To enable elasticity, microservices must be mobile and addressable. A Service Registry supports these requirements by maintaining an up-to-date list of available microservices that includes the host and port numbers for each instance. As load changes, the system can spawn or destroy instances in any location and continue to satisfy requests because the Service Registry maintains the addresses of all active instances for each microservice. 
+
+In order to register every microservice in the Service Registry there are two alternatives: self-registration and [3rd Party Registration](http://microservices.io/patterns/3rd-party-registration.html). When booting a Lagom microservice a registrar will register the name of the microservice, the URL and the names of the locatable Service Descriptors on the Service Registry so that they can be located. When powering down an instance of a service, the registrar will have to update the Service Registry too. The Service Registry must keep an up-to-date track of which instances can serve each service. Lagom's [[Developer Environment|DevEnvironment]] provides an implementation of a Service Registry and a registrar so you can run your microservices locally.
+
+Lagom supports deploying many Service Descriptors inside a single microservice. That is not against the recommendation that a [[microservice should only do one thing|Microservices#does-this-service-do-only-one-thing-]] and do it well. An example of such setup is adding a metrics service or an admin service alongside your domain service. This way when you start an instance of your microservice your infrastructure and your admin tools will be able to query and tune the microservice. You could also model your domain in a way that a single microservice contains multiple Service Descriptors. Following are the specifications of a `Greetings` microservice that contains two Service Descriptors (`welcome` and `farewell`) : 
```

Comment:
If we imagine that readers are going linearly through the documentation like a book, then this would appear before the idea of Service Descriptors has been introduced. To be honest, this paragraph and the following list seem out of place in this page, anyway. Maybe we should move them into the `ServiceDescriptors` page?